design/layer2-bind-interfaces.md

```diff
@@ -0,0 +1,243 @@
+The veth0 is belongs to subnet 192.168.1.0/24, the veth1 is belongs to subnet 192.172.1.0/24. When the Client(192.172.1.2)
+requests its LB Service(VIP is 192.172.1.10), it may send packets to veth0(due to the Speaker advertising VIP from all 
+interfaces), but K8s would reply on veth1, which led to asymmetric routing so that Linux drop the packets from the Client.
+The VIPs should listen to the suitable interfaces according to the network environment, other than all interfaces.
+
+[<img src="./images/complex_system.svg" width="800"/>](./images/complex_system.svg)
+
+Secondly, in a system that clearly distinguishes network planes, people hope to publish VIPs on the corresponding network planes 
+according to the LB Service function, rather than indiscriminately announcing VIPs on each network plane. For example, in the 
+following picture, if one LB Service is only requested by the `management Network`, we hope the Speaker only advertises this VIP 
+on eth0. Of course, this is not necessary.
+
+[<img src="./images/distinguish_network_plane.png" width="800"/>](./images/distinguish_network_plane.png)
+
+### Goals
+- In L2 mode, specify an interfaces list used by MetalLB to announce VIPs, applicable to all nodes and IPAddressPools.
+  (i.e. support config `(Interfaces)`)
+- In L2 mode, specify an interfaces list used by MetalLB to announce some specified IPAddressPools' IPs, applicable to all 
+  nodes. (i.e. support config `(IPAddressPools, Interfaces)`)
+- In L2 mode, specify an interfaces list used by MetalLB to announce some specified IPAddressPools' IPs, applicable to the 
+  specified nodes. (i.e. support config `(  IPAddressPools, Nodes, Interfaces)`)
+### Non-Goals
+- Determine the interface used to announce the VIP according to the host route.
+- Attempt to find and correct user's configuration errors. For example, if the user-configured interface's name does not exist, 
+  MetalLB will not announce the VIP instead of using another interface to announce the VIP.
+
+## Data model
+
+Based on the current MetalLB CRD design (`8d906f0abef3d7ed5b0435fa4edfbe5e37b38557`), we'd like to add a new field `Interfaces` to `L2Advertisement`.
+
+The new definition of `L2Advertisement` is:
+```go
+type L2Advertisement struct {
+  Name                     string `yaml:"name"`
+  IPAddressPools           []string `yaml:"ipAddressPools"`
+  IPAddressPoolSelector    *metav1.LabelSelector `yaml:"ipAddressPoolSelector"`
+  NodeSelector             *metav1.LabelSelector `yaml:"nodeSelector"`
+  Interfaces               []string `yaml:"interfaces"`
+}
+```
+- `Interfaces`: It's an interface name list. The VIP in IPAddressPools(specified by `IPAddressPools` or `IPAddressPoolSelector`) only be announced 
+  by the `Interfaces` of the nodes(specified by `NodeSelector`). If the field `IPAddressPools` and `IPAddressPoolSelector` 
+  is null, the `Interfaces` applicable to all IPAddressPools. If the field `NodeSelector` is null, the `Interfaces` 
```

Comment:
I'd remove the nodeselector paragraph, as it's part of the existing mechanism and nothing changes.